lib/jormungandr/src/Cardano/Wallet/Jormungandr.hs

```diff
@@ -128,31 +129,31 @@ serveWallet (cfg, sb, tr) databaseDir listen lj beforeMainLoop = do
             let nPort = Port $ baseUrlPort $ _restApi cp
             waitForService "Jörmungandr" (sb, tr) nPort $
                 waitForNetwork nl defaultRetryPolicy
-            rndApi <- apiLayer tr nl
-            seqApi <- apiLayer tr nl
-            startServer tr nPort nl rndApi seqApi
+            rndApi <- apiLayer tr (toWLBlock <$> nl)
+            seqApi <- apiLayer tr (toWLBlock <$> nl)
+            startServer tr nPort rndApi seqApi
             pure ExitSuccess
         Left e -> handleNetworkStartupError e
   where
     startServer
         :: Trace IO Text
         -> Port "node"
-        -> NetworkLayer IO t (Block Tx)
         -> ApiLayer (RndState t) t RndKey
         -> ApiLayer (SeqState t) t SeqKey
         -> IO ()
-    startServer tracer nPort nl rndWallet seqWallet = do
-        let (_, bp) = staticBlockchainParameters nl
+    startServer tracer nPort rndWallet seqWallet = do
         Server.withListeningSocket listen $ \(wPort, socket) -> do
             let tracerIPC = appendName "daedalus-ipc" tracer
             let tracerApi = appendName "api" tracer
             let settings = Warp.defaultSettings
-                    & setBeforeMainLoop (beforeMainLoop (Port wPort) nPort bp)
+                    & setBeforeMainLoop (beforeMainLoop (Port wPort) nPort)
             let ipcServer = daedalusIPC tracerIPC wPort
             let apiServer =
                     Server.start settings tracerApi socket rndWallet seqWallet
             race_ ipcServer apiServer
 
+    toWLBlock = J.convertBlock
```

Comment:
Intend to add `toSPBlock` right here.

> This function [toSPBlock] should probably live in the Binary module.

— @KtorZ  https://github.com/input-output-hk/cardano-wallet/pull/788#discussion_r333950088

That feels wrong to me. If `StakePool.Metrics` had a `Block`-datatype we'd force `Binary` to import `StakePool.Metrics`. I think it makes more sense to have the glue here (maybe as top level function though)

Reply:
We'll need deeper refactoring of the Jörmungandr package at some point. The structure is a bit unclear and we can do much better.